test/e2e/cluster-logging/sd_utils.go

```diff
@@ -57,75 +63,205 @@ func newGclLogsProvider(f *framework.Framework) (*gclLogsProvider, error) {
 		return nil, err
 	}
 
+	pubsubService, err := pubsub.New(hc)
+	if err != nil {
+		return nil, err
+	}
+
 	provider := &gclLogsProvider{
-		GclService: gclService,
-		Framework:  f,
+		GclService:         gclService,
+		PubsubService:      pubsubService,
+		Framework:          f,
+		LogEntryCache:      map[string]chan logEntry{},
+		CacheMutex:         &sync.Mutex{},
+		PollingStopChannel: make(chan struct{}, 1),
 	}
 	return provider, nil
 }
 
-func (logsProvider *gclLogsProvider) FluentdApplicationName() string {
-	return "fluentd-gcp"
+func (gclLogsProvider *gclLogsProvider) Init() error {
+	projectId := framework.TestContext.CloudConfig.ProjectID
+	nsName := gclLogsProvider.Framework.Namespace.Name
+
+	topic, err := gclLogsProvider.createPubSubTopic(projectId, nsName)
+	if err != nil {
+		return fmt.Errorf("failed to create PubSub topic: %v", err)
+	}
+	gclLogsProvider.Topic = topic
+
+	subs, err := gclLogsProvider.createPubSubSubscription(projectId, nsName, topic.Name)
+	if err != nil {
+		return fmt.Errorf("failed to create PubSub subscription: %v", err)
+	}
+	gclLogsProvider.Subscription = subs
+
+	logSink, err := gclLogsProvider.createGclLogSink(projectId, nsName, nsName, topic.Name)
+	if err != nil {
+		return fmt.Errorf("failed to create Stackdriver Logging sink: %v", err)
+	}
+	gclLogsProvider.LogSink = logSink
+
+	if err = gclLogsProvider.authorizeGclLogSink(); err != nil {
+		return fmt.Errorf("failed to authorize log sink: %v", err)
+	}
+
+	framework.Logf("Waiting for log sink to become operational")
+	time.Sleep(1 * time.Minute)
```

Comment:
I don't think there's a better way, since any solution wouldn't be able to reduce this delay to less than 10 sec, in reality probably more. Which IMO isn't worth additional complexity.